platform/features/layout/src/LayoutController.js

```diff
@@ -293,9 +344,102 @@ define(
          * view configuration.
          */
         LayoutController.prototype.endDrag = function () {
+            this.frameMoved = true;
+
+            setTimeout(function () {
+                this.frameMoved = false;
+            }.bind(this), 0);
+
             this.endDragInScope();
         };
 
+        /**
+         * Check if the object is currently selected.
+         *
+         * @param {string} obj the object to check for selection
+         * @returns {boolean} true if selected, otherwise false
+         */
+        LayoutController.prototype.selected = function (obj) {
+            return !!this.selectedId && this.selectedId === obj.getId();
+        };
+
+        /**
+         * Set the active user selection in this view.
+         *
+         * @param event the mouse event
+         * @param {string} id the object id
+         */
+        LayoutController.prototype.select = function (event, id) {
+            if (event) {
+                event.stopPropagation();
+                if (this.selection) {
+                    event.preventDefault();
+                }
+            }
+
+            var self = this;
+            var selectedObj = {};
+            var configuration = this.$scope.configuration;
+
+            this.selectedId = id;
+
+            // Toggle the visibility of the object frame
+            function toggle() {
+                // create new selection object so toolbar updates.
+                selectedObj = {};
+                if (!configuration.panels[id]) {
+                    configuration.panels[id] = {};
+                }
+
+                self.frames[id] = configuration.panels[id].hasFrame = !self.frames[id];
+
+                // Change which method is exposed, to influence
+                // which button is shown in the toolbar
+                selectedObj[self.frames[id] ? HIDE : SHOW] = toggle;
```

Comment:
@larkin I'll inline the names if moving the toggle() to the prototype works as suggested, otherwise will keep it as is.